ADAL/src/cache/ADTokenCacheItem.m

```diff
@@ -121,7 +121,8 @@ - (BOOL)isSameUser:(ADTokenCacheItem*) other
     
     if ([self isEmptyUser])
         return [other isEmptyUser];
-    return (nil != other.userInformation && [_userInformation.userId isEqualToString:other.userInformation.userId]);
+    
+    return [self.userInformation isEqual:other.userInformation];
```

Comment:
nit: please update comment for this method, currently:

`/*! Verifies if the user (as defined by userId) is the same between the two items. */`

That is not exactly true anymore, since now it will verify all the properties and not only userId.